### PR TITLE
[Keyboard] Add the Zed65 rev.1 PCB

### PR DESCRIPTION
**Pull request by mechlovin**
## Description

Add the Zed65 Rev. 1 PCB.
Thanks.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechlovin/zed65/rev1/rules.mk [hunk 1]
```diff
@@ -0,0 +1,3 @@
+
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
```
Comment (zvecr) [inline keyboards/mechlovin/zed65/rev1/rules.mk:2]:
```suggestion
```

File: keyboards/mechlovin/zed65/rev1/keymaps/default/readme.md
Comment (zvecr) [inline keyboards/mechlovin/zed65/rev1/keymaps/default/readme.md:1]: This file can be removed

File: keyboards/mechlovin/zed65/rev1/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    ),
+    [3] = LAYOUT_all(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                            KC_TRNS,                            KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+
```
Comment (zvecr) [inline keyboards/mechlovin/zed65/rev1/keymaps/via/keymap.c:50]:
```suggestion
    )
```

File: keyboards/mechlovin/zed65/rev1/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_all(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_DEL,  KC_HOME,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_PGUP,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGDN,
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_LSFT,          KC_UP,   KC_END,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, KC_RGUI, KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+
```
Comment (zvecr) [inline keyboards/mechlovin/zed65/rev1/keymaps/default/keymap.c:29]:
```suggestion
    )
```

File: keyboards/mechlovin/zed65/rev1/info.json [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    "url": "https://mechlovin.studio/", 
+    "maintainer": "mechlovin",
+    "keyboard_name": "Zed65 Rev1",
+    "usb": {
+        "vid": "0x4D4C",
+        "pid": "0x6505",
+        "device_version": "0.0.1"
+    },
+    "features": {
+        "backlight": false
```
Comment (zvecr) [inline keyboards/mechlovin/zed65/rev1/info.json:12]:
```suggestion
        "bootmagic": true,
        "command": false,
        "console": false,
        "extrakey": true,
        "mousekey": true,
        "nkro": true,
        "rgblight": true
```

File: keyboards/mechlovin/zed65/rev1/info.json [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    "matrix_pins": {
+        "rows": ["B12", "B13", "B14", "B15", "A1"],
+        "cols": ["B11", "B10", "B2", "B1", "B0", "A6", "A5", "A4", "A3", "C13", "B7", "B6", "B5", "B4", "B3"]
+    },
+    "indicators": {
+        "caps_lock": "B9",
+        "on_state": 0
+    },
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (zvecr) [inline keyboards/mechlovin/zed65/rev1/info.json:23]: Providing only `LAYOUT_all` is invalid

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/zed65/rev1/config.h [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#ifdef RGBLIGHT_ENABLE
+#define RGB_DI_PIN A7
```
Comment (drashna) [inline keyboards/mechlovin/zed65/rev1/config.h:21]:
```suggestion
#define WS2812_DI_PIN A7
```

File: keyboards/mechlovin/zed65/rev1/readme.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# zed65
+
+![Zed65](https://i.imgur.com/IbsSLb6.jpeg)
```
Comment (drashna) [inline keyboards/mechlovin/zed65/rev1/readme.md:3]:
```suggestion
![Zed65](https://i.imgur.com/IbsSLb6h.jpeg)
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/mechlovin/zed65/rev1/config.h [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+#    define RGBLIGHT_EFFECT_BREATHING
+#    define RGBLIGHT_EFFECT_RAINBOW_MOOD
+#    define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#    define RGBLIGHT_EFFECT_SNAKE
+#    define RGBLIGHT_EFFECT_KNIGHT
+#    define RGBLIGHT_EFFECT_CHRISTMAS
+#    define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#    define RGBLIGHT_EFFECT_RGB_TEST
+#    define RGBLIGHT_EFFECT_ALTERNATING
+#    define RGBLIGHT_EFFECT_TWINKLE
```
Comment (waffle87) [inline keyboards/mechlovin/zed65/rev1/config.h:46]:
```suggestion
#define WS2812_SPI SPID1 // default: SPID1
#define WS2812_SPI_MOSI_PAL_MODE 6 // MOSI pin "alternate function", see the respective datasheet for the appropriate values for your MCU. default: 5
#define WS2812_SPI_USE_CIRCULAR_BUFFER
```

File: keyboards/mechlovin/zed65/rev1/info.json [hunk 9]
```diff
@@ -0,0 +1,332 @@
...
+        "rgblight": true
+    },
+    "matrix_pins": {
+        "rows": ["B12", "B13", "B14", "B15", "A1"],
+        "cols": ["B11", "B10", "B2", "B1", "B0", "A6", "A5", "A4", "A3", "C13", "B7", "B6", "B5", "B4", "B3"]
+    },
+    "indicators": {
+        "caps_lock": "B9",
+        "on_state": 0
+    },
```
Comment (waffle87) [inline keyboards/mechlovin/zed65/rev1/info.json:22]:
```suggestion
    },
    "rgblight": {
        "pin": "A7",
        "led_count": 24,
        "sleep": true,
        "animations": {
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
        }
    },
```

File: keyboards/mechlovin/zed65/rev1/info.json [hunk 10]
```diff
@@ -0,0 +1,332 @@
...
+    "layouts": {
```
Comment (waffle87) [inline keyboards/mechlovin/zed65/rev1/info.json:23]:
```suggestion
    "community_layouts": ["65_ansi_blocker_tsangan"],
    "layouts": {
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/mechlovin/zed65/rev1/info.json [hunk 11]
```diff
@@ -0,0 +1,350 @@
...
+            "christmas": true,
+            "knight": true,
+            "rainbow_mood": true,
+            "rainbow_swirl": true,
+            "rgb_test": true,
+            "snake": true,
+            "static_gradient": true,
+            "twinkle": true
+        }
+    },
```
Comment (waffle87) [inline keyboards/mechlovin/zed65/rev1/info.json:39]:
```suggestion
    },
    "ws2812": {
        "driver": "spi"
    },
```

File: keyboards/mechlovin/zed65/rev1/rules.mk [hunk 12]
```diff
@@ -0,0 +1 @@
+WS2812_DRIVER = spi
```
Comment (waffle87) [inline keyboards/mechlovin/zed65/rev1/rules.mk:1]:
```suggestion
# This file intentionally left blank
```

**Review by waffle87 — COMMENTED** [review]
Sorry, forgot a couple.

**Review by waffle87 — APPROVED** [review]

**Comment by keyboard-magpie** [comment]
Failing CI on missing diode directions- @mechlovin is this row2col or col2row

**Comment by mechlovin** [comment]
col2row on zed65/config.h.
Should I move it from config.h to info.json?

**Comment by keyboard-magpie** [comment]
@mechlovin thanks for prompt response, will need to be in info.json, seems to have been overlooked.

Needs a line like `  "diode_direction": "COL2ROW",` 

Will need a new PR if possible please as bugfix.

**Comment by mechlovin** [comment]
I will. thanks.

**Pull request merged**